std/typecons.d

```diff
@@ -3445,7 +3445,7 @@ Returns:
         }
         //Need to use 'is' if T is a float type
         //because NaN != NaN
-        else static if (isFloatingPoint!T)
+        else static if (__traits(compiles, { static assert(!(nullValue == nullValue)); }))
```

Comment:
Is there no simpler way to write this?

Reply:
I wrote that instead of `else static if (nullValue != nullValue)` in case `opEquals` isn't CTFE-able. Is it okay if that causes compilation to fail?

Reply:
I figured that was the reason. Unfortunately I don't think we have something like `if (CanExecAtCompileTime!(expr)`.

The problem is with requiring T.opEqual to be CTFE-able. I don't have a lot of time to think about it, but my first instinct would be to move the check to runtime time in this case.

```
...
else static if (isFloatingPoint!T) { ...}
else
{
    if (nullValue != nullValue)
        return _value is nullValue;
    else
        return _value == nullValue;
}
```

Though that's obviously not ideal.

Reply:
I think it would be sufficient to add a unittest inside the template along the lines of:

```d
@system unittest
{
    assert(typeof(this)(nullValue).isNull, typeof(this).stringof ~ 
        ".isNull does not work correctly because " ~ typeof(T).stringof ~
        " has an == operator that is non-reflexive and could not be " ~
        " determined before runtime to be non-reflexive!");
}
```